### PR TITLE
tests: rather course (but it works) test coverage for session creation.

### PR DESCRIPTION
**Pull request by JohnMcLear**
*will fail intentionally*

In response to https://github.com/ether/etherpad-lite/issues/4898

### CODE REVIEW

File: src/tests/backend/specs/api/httpRouteSessionCreation.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  '/',
+  '/api/',
+  '/favicon.ico',
+  '/locales.json',
+  '/pluginfw/plugin-definitions.json',
+  '/static/js/pad.js',
+  '/stats/',
+];
+
+const getDatabaseSize = () => {
```
Comment (rhansen) [inline src/tests/backend/specs/api/httpRouteSessionCreation.js:23]: This is fragile. Instead, implement the [`Store.length() method`](https://www.npmjs.com/package/express-session#storeallcallback) in the SessionStore class in `SessionStore.js`.

File: src/tests/backend/specs/api/httpRouteSessionCreation.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    if (settings.dbType !== 'dirty') this.skip;
+
+    this.timeout(100);
+
+    for (const endpoint of shouldNotCreateSession) {
+      it(endpoint, async function () {
+        const previousCount = getDatabaseSize();
+        await agent.get(endpoint)
+        .expect(200)
+        .expect(() => {
```
Comment (rhansen) [inline src/tests/backend/specs/api/httpRouteSessionCreation.js:42]: Also test that the server does not reply with a `express_sid` cookie.

File: src/tests/backend/specs/api/httpRouteSessionCreation.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        })
+      });
+    }
+
+    for (const endpoint of shouldCreateSession) {
+      const previousCount = getDatabaseSize();
+      it(endpoint, async function () {
+        await agent.get(endpoint)
+        .expect(200)
+        .expect(() => {
```
Comment (rhansen) [inline src/tests/backend/specs/api/httpRouteSessionCreation.js:54]: Also test that the server replies with an `express_sid` cookie.

**Comment by webzwo0i** [comment]
just so we don't forget them: atm also any 404s will cause a session creation/cookie sent

File: src/tests/backend/specs/api/httpRouteSessionCreation.js (hunk 1 above)
Comment (JohnMcLear) [reply src/tests/backend/specs/api/httpRouteSessionCreation.js]: I made an attempt at this but it didn't work.  I'm unsure how to access ``Store`` as it's the parent of ``SessionStore`` (which is extending ``Store``)

File: src/tests/backend/specs/api/httpRouteSessionCreation.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+const common = require('../../common');
+const settings = require('../../../../node/utils/Settings');
+const fs = require('fs');
+let agent;
+
+const shouldCreateSession = [
+  '/p/foo',
+  '/p/foo/export/html',
+];
+const shouldNotCreateSession = [
```
[the remark this answers is not in the document]
Comment (JohnMcLear) [reply src/tests/backend/specs/api/httpRouteSessionCreation.js]: Done :)

File: src/tests/backend/specs/api/httpRouteSessionCreation.js (hunk 3 above)
Comment (JohnMcLear) [reply src/tests/backend/specs/api/httpRouteSessionCreation.js]: Done

File: src/tests/backend/specs/api/httpRouteSessionCreation.js (hunk 1 above)
Comment (webzwo0i) [reply src/tests/backend/specs/api/httpRouteSessionCreation.js]: If I read it correctly, we would need to add our own `length` method and increase/decrease it on set/destroy

**Comment by JohnMcLear** [comment]
Oh but the link shows a method already exists for length...??

**Comment by webzwo0i** [comment]
Yep, but we don't implement `Store.all` which is needed to get all the active sessions (that's needed by `length`). My statement was not quite correct: As per @rhansen link we would need a method `all` that returns all active sessions as an array. Thus we would not need to implement our own increase/decrease.

**Comment by JohnMcLear** [comment]
Ack, no problems, will try sort that before I do proxy stuff 👍🏽

**Comment by JohnMcLear** [comment]
Wont ``sessions.length`` only include most recent and not persist?  Else we would need to get all at startup...  It's fine if it is since startup but it's gonna be confusing if someone else uses it....

File: src/tests/backend/specs/api/httpRouteSessionCreation.js (hunk 2 above)
Comment (JohnMcLear) [reply src/tests/backend/specs/api/httpRouteSessionCreation.js]: I think this is done, but might need polish.

**Comment by JohnMcLear** [comment]
I had a quick stab at this but I feel like I'm on the wrong track so will wait for input from @rhansen before spending any more time :)

**Pull request closed without merging**